internal/supervisor/common.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package supervisor
+
+import (
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type clusterInstallationClaimStore interface {
+	GetClusterInstallations(*model.ClusterInstallationFilter) ([]*model.ClusterInstallation, error)
+	clusterInstallationLockStore
+}
+
+func claimClusterInstallation(store clusterInstallationClaimStore, installation *model.Installation, instanceID string, logger log.FieldLogger) (*model.ClusterInstallation, *clusterInstallationLock, error) {
+	clusterInstallationFilter := &model.ClusterInstallationFilter{
+		InstallationID: installation.ID,
+		Paging:         model.AllPagesNotDeleted(),
+	}
+	clusterInstallations, err := store.GetClusterInstallations(clusterInstallationFilter)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Failed to get cluster installations")
```

Comment:
super nit: lowercase letter on wrapped error returns
```suggestion
		return nil, nil, errors.Wrap(err, "failed to get cluster installations")
```